src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -106,22 +106,33 @@
             transaction {:id         id
                          :from       from
                          :to         to
+                         :to-name    (when (= to constants/contract-address)
+                                       (i18n/label :t/new-contract))
                          :symbol     symbol
                          :value      (money/bignumber (or value 0))
                          :data       data
-                         :gas        (money/bignumber (money/to-decimal gas))
-                         :gas-price  (money/bignumber (money/to-decimal gasPrice))
+                         :gas        (when (seq gas)
+                                       (money/bignumber (money/to-decimal gas)))
+                         :gas-price  (when (seq gasPrice)
+                                       (money/bignumber (money/to-decimal gasPrice)))
                          :timestamp  now
                          :message-id message_id}
-            sending-from-chat? (not (get-in db [:wallet :send-transaction :waiting-signal?]))
+            sending-from-bot-or-dapp? (not (get-in db [:wallet :send-transaction :waiting-signal?]))
             new-db (assoc-in db [:wallet :transactions-unsigned id] transaction)
             sending-db {:id         id
-                        :from-chat? sending-from-chat?}]
-        (if sending-from-chat?
-          ;;SENDING FROM CHAT
-          {:db       (assoc-in new-db [:wallet :send-transaction] sending-db) ; we need to completely reset sending state here
-           :dispatch-n [[:update-wallet (map :symbol (tokens/tokens-for (ethereum/network->chain-keyword (:network db))))] [:navigate-to-modal :wallet-send-transaction-modal]]}
-          ;;SEND SCREEN WAITING SIGNAL
+                        :from-chat? sending-from-bot-or-dapp?}]
+        (if sending-from-bot-or-dapp?
+          ;;SENDING FROM BOT (CHAT) OR DAPP
+          {:db         (assoc-in new-db [:wallet :send-transaction] sending-db) ; we need to completely reset sending state here
+           :dispatch-n (into []
```

Comment:
Maybe we can have some helper function for this? Looks like we frequently do something similar.
Or does `dispatch-n` ignore nils? If so then what about 
```
[[:navigate-to-modal :wallet-send-transaction-modal]
 (when-not (seq gas)
  [:wallet/update-estimated-gas transaction])]
```

Reply:
oh, i think you're right, i saw they fixed `nil` in `dispatch-n`

Reply:
thanks @jeluard , fixed